src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
+    }
+    return payload;
```

Comment:
PullProofAI

There is a potential issue with the error handling in the `validate` method of the `JwtStrategy` class. If an error occurs while retrieving the JWT token from Redis, it is not being properly handled, which could lead to uncaught exceptions and potentially unexpected behavior.

### Proposed Improvement

To address this issue, it's recommended to add proper error handling when fetching the JWT token from Redis. You can catch any errors that may occur during this operation and handle them appropriately, such as logging the error or throwing a custom exception to provide more context.

```typescript
---new_hunk---
21:     const jwtToken = await client.get(`jwt:${userId}`);
22:     if (!jwtToken) {
23:       throw new UnauthorizedException();
24:     }
25:     return payload;
26:   } catch (error) {
27:     // Handle the error here, e.g., log the error or throw a custom exception
28:     throw new Error('Failed to retrieve JWT token from Redis');
29:   }
```

By adding proper error handling, you ensure that any issues with fetching the JWT token are appropriately dealt with, improving the robustness and reliability of the authentication process.


<!-- This is an auto-generated comment by PullProofAI -->